### PR TITLE
planner: add variables for instance plan cache

### PR DESCRIPTION
**Pull request by qw4990**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions))

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/18830
- Other reference link(s): https://github.com/pingcap/tidb/issues/54057

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -2966,6 +2979,32 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+
+> **警告：**
+>
+> Instance Plan Cache 目前是实验性功能，不建议在生产环境中使用。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`125829120`120 MiB）
```
Comment (time-and-fate) [inline system-variables.md:3005]: 缺个括号

Comment (qw4990) [inline system-variables.md:3005]:
```suggestion
- 默认值：`125829120`（120 MiB）
```

File: system-variables.md [hunk 2]
```diff
@@ -2966,6 +2979,32 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+> **警告：**
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`104857600`（即 100 MiB）
+- TiDB 后台定期对 Instance Plan Cache 进行清理，这个变量控制 Cache 的目标内存，也就是说 TiDB 每次会把 Plan Cache 清理到这个变量的内存大小。
```
Comment (Oreoxmt) [inline system-variables.md:2993]:
```suggestion
- 这个变量用于设置 Instance Plan Cache 的目标内存大小。TiDB 后台定期清理 Instance Plan Cache，以确保其内存占用不超过该变量设置的值。
```

File: system-variables.md [hunk 3]
```diff
@@ -2180,6 +2180,19 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
+> **警告：**
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量用于是否开启 Instance Plan Cache 功能。
```
Comment (Oreoxmt) [inline system-variables.md:2194]:
```suggestion
- 这个变量用于控制是否开启 Instance Plan Cache 功能。该功能实现实例级执行计划缓存，允许同一个 TiDB 实例的所有会话共享执行计划缓存，从而提升内存利用率。
```

File: system-variables.md [hunk 4]
```diff
@@ -2966,6 +2979,32 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+> **警告：**
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`125829120`（即 120 MiB）
+- 这个变量用于设置 Instance Plan Cache 的最大内存使用量。该值必须大于 [`tidb_instance_plan_cache_target_mem_size`](#tidb_instance_plan_cache_target_mem_size-span-classversion-mark从-v840-版本开始引入span)。
```
Comment (Oreoxmt) [inline system-variables.md:3006]:
```suggestion
- 这个变量用于设置 Instance Plan Cache 的最大内存使用量。该值必须大于 [`tidb_instance_plan_cache_target_mem_size`](#tidb_instance_plan_cache_target_mem_size-v840-版本开始引入)。
```

**Comment by songrijie** [comment]
这3个变量要说明下单位是“字节”。

File: system-variables.md [hunk 5]
```diff
@@ -2966,6 +2979,32 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+> **警告：**
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`125829120`（即 120 MiB）
+- 这个变量用于设置 Instance Plan Cache 的最大内存使用量。该值必须大于 [`tidb_instance_plan_cache_target_mem_size`](#tidb_instance_plan_cache_target_mem_size-v840-版本开始引入)。
```
Comment (qw4990) [inline system-variables.md:3006]:
```suggestion
### `tidb_instance_plan_cache_max_size` <span class="version-mark">从 v8.4.0 版本开始引入</span>

> **警告：**
>
> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。

- 作用域：GLOBAL
- 是否持久化到集群：是
- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
- 类型：整数型
- 默认值：`125829120`（即 120 MiB）
- 这个变量用于设置 Instance Plan Cache 的最大内存使用量。
```

File: system-variables.md [hunk 6]
```diff
@@ -2966,6 +2979,32 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+> **警告：**
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`104857600`（即 100 MiB）
+- 这个变量用于设置 Instance Plan Cache 的目标内存大小。TiDB 后台定期清理 Instance Plan Cache，以确保其内存占用不超过该变量设置的值。
```
Comment (qw4990) [inline system-variables.md:2993]:
```suggestion
### tidb_instance_plan_cache_reserved_percentage  <span class="version-mark">从 v8.4.0 版本开始引入</span>

> **警告：**
>
> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。

- 作用域：GLOBAL
- 是否持久化到集群：是
- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
- 类型：浮点型
- 默认值：`0.1`
- 范围：`[0, 1]`
- 这个变量控制每次逐出的内存大小。TiDB 会在 Plan Cache 满后，也就是内存达到 `tidb_instance_plan_cache_max_size` 开始逐出。每次逐出 `tidb_instance_plan_cache_max_size * tidb_instance_plan_cache_reserved_percentage` 数量的内存。
```

File: system-variables.md [hunk 7]
```diff
@@ -2180,6 +2180,19 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
+> **警告：**
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量用于控制是否开启 Instance Plan Cache 功能。该功能实现实例级执行计划缓存，允许同一个 TiDB 实例的所有会话共享执行计划缓存，从而提升内存利用率。
```
Comment (songrijie) [inline system-variables.md:2194]:
```suggestion
- 这个变量用于控制是否开启 Instance Plan Cache 功能。该功能实现实例级执行计划缓存，允许同一个 TiDB 实例的所有会话共享执行计划缓存，从而提升内存利用率。 在开启此功能之前，建议关闭实例级执行计划缓存，具体方法参见 [Prepare 语句执行计划缓存](/sql-prepared-plan-cache.md) 和 [非 Prepare 语句执行计划缓存](/sql-non-prepared-plan-cache.md)。
```

File: system-variables.md [hunk 8]
```diff
@@ -2966,6 +2979,33 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：浮点型
+- 默认值：`0.1`
+- 范围：`[0, 1]`
+- 这个变量控制每次逐出的内存大小。TiDB 会在 Plan Cache 满后，也就是内存达到 `tidb_instance_plan_cache_max_size` 开始逐出。每次逐出 `tidb_instance_plan_cache_max_size * tidb_instance_plan_cache_reserved_percentage` 数量的内存。
```
Comment (songrijie) [inline system-variables.md:2994]:
```suggestion
- 这个变量控制发生内存驱逐动作后，空闲内存的百分比。TiDB 会在 Plan Cache 使用的内存达到 `tidb_instance_plan_cache_max_size` 时，依据 LRU 开始驱逐内存中的执行计划，直到空闲内存比例超过 `tidb_instance_plan_cache_reserved_percentage` 的定义为止。
```

File: system-variables.md [hunk 9]
```diff
@@ -2966,6 +2979,33 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+> **警告：**
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`125829120`（即 120 MiB）
+- 这个变量用于设置 Instance Plan Cache 的最大内存使用量。
```
Comment (songrijie) [inline system-variables.md:3007]:
```suggestion
- 这个变量用于设置 Instance Plan Cache 的最大内存使用量，单位是字节。
```

File: system-variables.md [hunk 10]
```diff
@@ -2180,6 +2180,19 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
+> **警告：**
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量用于控制是否开启 Instance Plan Cache 功能。该功能实现实例级执行计划缓存，允许同一个 TiDB 实例的所有会话共享执行计划缓存，从而提升内存利用率。 在开启此功能之前，建议关闭实例级执行计划缓存，具体方法参见 [Prepare 语句执行计划缓存](/sql-prepared-plan-cache.md) 和 [非 Prepare 语句执行计划缓存](/sql-non-prepared-plan-cache.md)。
```
Comment (Oreoxmt) [inline system-variables.md:2194]:
```suggestion
- 这个变量用于控制是否开启 Instance Plan Cache 功能。该功能实现实例级执行计划缓存，允许同一个 TiDB 实例的所有会话共享执行计划缓存，从而提升内存利用率。开启该功能之前，建议关闭会话级别的 [Prepare 语句执行计划缓存](/sql-prepared-plan-cache.md)和[非 Prepare 语句执行计划缓存](/sql-non-prepared-plan-cache.md)。
```

File: system-variables.md [hunk 11]
```diff
@@ -2966,6 +2979,33 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
 - 单位：行
 - 这个变量用来设置执行过程中初始 chunk 的行数。默认值是 32，可设置的范围是 1～32。chunk 行数直接影响单个查询所需的内存。可以按照查询中所有的列的总宽度和 chunk 行数来粗略估算单个 chunk 所需内存，并结合执行器的并发数来粗略估算单个查询所需内存总量。建议单个 chunk 内存总量不要超过 16 MiB。
 
+### tidb_instance_plan_cache_reserved_percentage  <span class="version-mark">从 v8.4.0 版本开始引入</span>
```
Comment (Oreoxmt) [inline system-variables.md:2982]:
```suggestion
### `tidb_instance_plan_cache_reserved_percentage` <span class="version-mark">从 v8.4.0 版本开始引入</span>
```

File: system-variables.md [hunk 12]
```diff
@@ -2966,6 +2979,33 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：浮点型
+- 默认值：`0.1`
+- 范围：`[0, 1]`
+- 这个变量控制发生内存驱逐动作后，空闲内存的百分比。TiDB 会在 Plan Cache 使用的内存达到 `tidb_instance_plan_cache_max_size` 时，依据 LRU 开始驱逐内存中的执行计划，直到空闲内存比例超过 `tidb_instance_plan_cache_reserved_percentage` 的定义为止。
```
Comment (Oreoxmt) [inline system-variables.md:2994]:
```suggestion
- 这个变量用于控制内存驱逐后 Instance Plan Cache 的空闲内存百分比。当 Instance Plan Cache 使用的内存达到 [`tidb_instance_plan_cache_max_size`](#tidb_instance_plan_cache_max_size从-v840-版本开始引入) 设置的上限时，TiDB 会按照 Least Recently Used (LRU) 算法开始驱逐内存中的执行计划，直到空闲内存比例超过 [`tidb_instance_plan_cache_reserved_percentage`](#tidb_instance_plan_cache_reserved_percentage-从-v840-版本开始引入) 设定的值。```

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 13]
```diff
@@ -2966,6 +2979,34 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：浮点型
+- 默认值：`0.1`
+- 范围：`[0, 1]`
+- 这个变量用于控制内存驱逐后 Instance Plan Cache 的空闲内存百分比。当 Instance Plan Cache 使用的内存达到 [`tidb_instance_plan_cache_max_size`](#tidb_instance_plan_cache_max_size从-v840-版本开始引入) 设置的上限时，TiDB 会按照 Least Recently Used (LRU) 算法开始驱逐内存中的执行计划，直到空闲内存比例超过 [`tidb_instance_plan_cache_reserved_percentage`](#tidb_instance_plan_cache_reserved_percentage-从-v840-版本开始引入) 设定的值。
```
Comment (Oreoxmt) [inline system-variables.md:2994]:
```suggestion
- 这个变量用于控制内存驱逐后 Instance Plan Cache 的空闲内存百分比。当 Instance Plan Cache 使用的内存达到 [`tidb_instance_plan_cache_max_size`](#tidb_instance_plan_cache_max_size-从-v840-版本开始引入) 设置的上限时，TiDB 会按照 Least Recently Used (LRU) 算法开始驱逐内存中的执行计划，直到空闲内存比例超过 [`tidb_instance_plan_cache_reserved_percentage`](#tidb_instance_plan_cache_reserved_percentage-从-v840-版本开始引入) 设定的值。
```

File: system-variables.md [hunk 14]
```diff
@@ -2966,6 +2979,34 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：浮点型
+- 默认值：`0.1`
+- 范围：`[0, 1]`
+- 这个变量用于控制内存驱逐后 Instance Plan Cache 的空闲内存百分比。当 Instance Plan Cache 使用的内存达到 [`tidb_instance_plan_cache_max_size`](#tidb_instance_plan_cache_max_size-从-v840-版本开始引入) 设置的上限时，TiDB 会按照 Least Recently Used (LRU) 算法开始驱逐内存中的执行计划，直到空闲内存比例超过 [`tidb_instance_plan_cache_reserved_percentage`](#tidb_instance_plan_cache_reserved_percentage-从-v840-版本开始引入) 设定的值。
```
Comment (qiancai) [inline system-variables.md:2994]:
```suggestion
- 这个变量用于控制内存驱逐后 [Instance Plan Cache](/system-variables.md#tidb_enable_instance_plan_cache-从-v840-版本开始引入) 的空闲内存百分比。当 Instance Plan Cache 使用的内存达到 [`tidb_instance_plan_cache_max_size`](#tidb_instance_plan_cache_max_size-从-v840-版本开始引入) 设置的上限时，TiDB 会按照 Least Recently Used (LRU) 算法开始驱逐内存中的执行计划，直到空闲内存比例超过 [`tidb_instance_plan_cache_reserved_percentage`](#tidb_instance_plan_cache_reserved_percentage-从-v840-版本开始引入) 设定的值。
```

File: system-variables.md [hunk 15]
```diff
@@ -2966,6 +2979,34 @@ v5.0 后，用户仍可以单独修改以上系统变量（会有废弃警告）
...
+>
+> Instance Plan Cache 目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 是否受 Hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value) 控制：否
+- 类型：整数型
+- 默认值：`125829120`（即 120 MiB）
+- 单位：字节
+- 这个变量用于设置 Instance Plan Cache 的最大内存使用量。
```
Comment (qiancai) [inline system-variables.md:3008]:
```suggestion
- 这个变量用于设置 [Instance Plan Cache](/system-variables.md#tidb_enable_instance_plan_cache-从-v840-版本开始引入) 的最大内存使用量。
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

**Pull request merged**